### DISCUSSION TITLE
FrePPLe search for demand fulfillment for 5 years when executing constraint planning

### DISCUSSION THREAD

**Discussion by gibhsu** [frePPLe/frepple#482] [Q&A] [answered]
I have a demand 'UA9421015200_2_1_W' and run in Constraint plan type without any WIP and item supplier for raw material.
The purpose is to verify if FrePPLE can look for all WIP from all possible routes for the first run.
However, we faced the performance issue and tried to look into the logs.
In the logs we found this case that the order due date is 2021-06-09 2AM (yyyy-mm-dd), and FrePPLE was looking for supply until 2026-06-08T02:00:00.
I believe it was searching the supply for 5 years. Is it because I didn't configure the calendar for planning horizon.

Logs:
Start solving cluster 2
Planning demand 'UA9421015200_2_1_W': 1, 2021-06-09T02:00:00, 25
Demand 'UA9421015200_2_1_W' asks: 25 2021-06-09T02:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2021-06-13T02:00:00 0 100
Demand 'UA9421015200_2_1_W' asks: 25 2021-06-13T02:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2021-06-13T03:00:00 0 15854
Demand 'UA9421015200_2_1_W' asks: 25 2021-06-13T03:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2021-06-13T04:00:00 0 31608
Demand 'UA9421015200_2_1_W' asks: 25 2021-06-13T04:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2021-06-13T05:00:00 0 47362
Demand 'UA9421015200_2_1_W' asks: 25 2021-06-13T05:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2021-06-13T06:00:00 0 63116
Demand 'UA9421015200_2_1_W' asks: 25 2021-06-13T06:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2021-06-13T07:00:00 0 78870
Demand 'UA9421015200_2_1_W' asks: 25 2021-06-13T07:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2021-06-13T08:00:00 0 94624
Demand 'UA9421015200_2_1_W' asks: 25 2021-06-13T08:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2021-06-13T09:00:00 0 110378
Demand 'UA9421015200_2_1_W' asks: 25 2021-06-13T09:00:00

.....

Demand 'UA9421015200_2_1_W' asks: 25 2026-06-07T23:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2026-06-08T00:00:00 0 5.70148e+07
Demand 'UA9421015200_2_1_W' asks: 25 2026-06-08T00:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2026-06-08T01:00:00 0 5.70158e+07
Demand 'UA9421015200_2_1_W' asks: 25 2026-06-08T01:00:00
Demand 'UA9421015200_2_1_W' gets answer: 0 2026-06-08T02:00:00 0 5.70168e+07

**Comment by jdetaeye** [comment, marked as answer]
A couple of remarks on this;
- If with each attempt the answer is a delay, there is something in your model that makes the demand unplannable.  Run with loglevel>1 to see more detail. That's the root cause, and the other remarks below are not addressing the root cause but only reducing the impact of the mistake.
- The default max-lateness of an order is 5 years.   You can explicitly fill in another duration in the demand table.
- Add a parameter "plan.minimumdelay" with value "86400" to jump with increments of 1 day rather than the default of 1 hour.

**Reply by gibhsu** [reply]
I change the parameter  "plan.minimumdelay" to 86400 today. The performance did improve a lot. Thanks. If I have different time bucket for different period.For example, first 3 months using daily bucket, after that using weekly bucket. Then I should configure the plan.calendar correct?

**Comment by jdetaeye** [comment]
Do NOT set the the plan.calendar parameter!  
It's used only for exceptional cases where manufacturing orders & purchase order explicitly need to align to certain calendar.   

The planning algorithm in frepple isn't dividing the plan in time buckets. It uses a continuous representation of time.
Time buckets are used only for the visualization / aggregation of plan results.
